File: implementation/oneandtwo.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
  int tt;
  cin>>tt;
  while(tt--){
    int n;
  cin>>n;
  int flag=1;
  vector<int>v(n);
  int two=0;
  for (int i = 0; i < n; i++)
  {
    cin>>v[i];
    if(v[i]==2){
        two++;
    }
  }
  int two2=0;
  for (int i = 0; i < n; i++)
  {
    
    if(v[i]==2){
        two2++;
    }
    if(two2*2==two){
        cout<<i+1<<endl;
        flag=0;
        break;
    }
    

  } 
  if(flag==1){
    cout<<-1<<endl;
  } 
  }
  return 0;
}

File: python/oneandtwo.py
tt=int(input())
for _ in range(tt):
    n=int(input())
    a=list(map(int, input().split()))
    two=0
    f=1
    for i in a:
        if(i==2):
            two+=1
    two2=0
    for i in range(n):
        if(a[i]==2):
            two2+=1
        if(two2*2==two):
            print(i+1)
            f=0
            
            break
    if(f==1):
        print(-1)     